### ISSUE TITLE
[REVIEW]: StoSpa2: A C++ software package for stochastic simulations of spatially extended systems

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2293]
**Submitting author:** @BartoszBartmanski (<a href="http://orcid.org/0000-0002-0495-9150">Bartosz Bartmanski</a>)
**Repository:** <a href="https://github.com/BartoszBartmanski/StoSpa2" target ="_blank">https://github.com/BartoszBartmanski/StoSpa2</a>
**Version:** 2.0.29
**Editor:** @pdebuyl
**Reviewers:** @CFGrote, @mbkumar
**Archive:** <a href="https://doi.org/10.5281/zenodo.3901670" target="_blank">10.5281/zenodo.3901670</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/230fc08d3b1827d382b60493909e73f7/status.svg)](https://joss.theoj.org/papers/230fc08d3b1827d382b60493909e73f7)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/230fc08d3b1827d382b60493909e73f7"><img src="https://joss.theoj.org/papers/230fc08d3b1827d382b60493909e73f7/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/230fc08d3b1827d382b60493909e73f7/status.svg)](https://joss.theoj.org/papers/230fc08d3b1827d382b60493909e73f7)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/BartoszBartmanski/StoSpa2" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@CFGrote and @mbkumar , please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @pdebuyl know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/BartoszBartmanski/StoSpa2">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@BartoszBartmanski) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/BartoszBartmanski/StoSpa2">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@BartoszBartmanski) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1371/journal.pcbi.1004923 is OK
- 10.1063/1.4975167 is OK
- 10.1371/journal.pcbi.1005387 is OK
- 10.1007/s40571-015-0082-3 is OK
- 10.1063/1.4816377 is OK

MISSING DOIs

- https://doi.org/10.1137/070705039 may be missing for title: The reaction-diffusion master equation as an asymptotic approximation of diffusion to a small target

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02293/joss.02293/10.21105.joss.02293.pdf)

**Comment by pdebuyl** [comment]
@whedon add @mbkumar as reviewer

**Comment by whedon** [comment]
OK, @mbkumar is now a reviewer

**Comment by pdebuyl** [comment]
@CFGrote , @mbkumar , make sure to accept the invitation to the reviewers group and to have a look at the [reviewer guidelines](https://joss.readthedocs.io/en/latest/reviewer_guidelines.html).

The review process will happen in this issue page, so questions to the author or to me can be added as comments here.

**Comment by pdebuyl** [comment]
@mbkumar please write the comments on the manuscript here, so that the review page keeps trace of it. The issues in the software repository are more suited for direct software problems (bugs, missing dependencies, for instance).

**Comment by mbkumar** [comment]
Section on the status of the field is missing.

**Comment by mbkumar** [comment]
Completed the review and my recommendation is publish as it is. 

The manuscript is well written and the software was coded well with best practices.

@pdebuyl 
Could you please send me an email acknowledging the completion of review for my record?

**Comment by CFGrote** [comment]
@whedon check repository

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.84  T=0.19 s (254.1 files/s, 121196.1 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
C/C++ Header                     8           3287           1366          13960
SVG                              1              0             16           1014
Markdown                         3            100             10            446
Python                          11            133             87            396
C++                              7             95             26            276
Jupyter Notebook                 1              0            651            259
YAML                             2             38             13            173
TeX                              1             10              0            130
reStructuredText                 4             90             24            123
CMake                            7             32             19             83
Bourne Shell                     2              7              4             26
TOML                             1              0              0              2
-------------------------------------------------------------------------------
SUM:                            48           3792           2216          16888
-------------------------------------------------------------------------------


Statistical information for the repository '2293' was gathered on 2020/06/15.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
bartosz.bartmanski              43         21078            642          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
bartosz.bartmanski        20436           97.0          0.4                8.79
```

**Comment by CFGrote** [comment]
Can't run the example. See https://github.com/BartoszBartmanski/StoSpa2/issues/8 .

**Comment by CFGrote** [comment]
The target audience is not clearly mentioned (neither in README nor in the documentation).

**Comment by CFGrote** [comment]
I suggest to add short sections on how to contribute and how to seek support (i.e. link to gh issues) to the README.

**Comment by CFGrote** [comment]
Some references in the manuscript are not correctly rendered in the autogenerated pdf but this may be a conversion problem. Please check the page number in Osborne et al 2017) and Isaacson 2009.

**Comment by CFGrote** [comment]
Installation instructions:
Only after reading the manuscript, I understood that it is not required to compile the c++ code if one is only using the python interface. This should maybe be stated more clearly in the online installation instructions in README and docs.

**Comment by pdebuyl** [comment]
@mbkumar Thank you for the review (email sent).

**Comment by BartoszBartmanski** [comment]
@CFGrote thank you for these comments. I just pushed two commits to the repository for StoSpa2, which hopefully address all of the above mentioned points.

**Comment by BartoszBartmanski** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02293/joss.02293/10.21105.joss.02293.pdf)

**Comment by BartoszBartmanski** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02293/joss.02293/10.21105.joss.02293.pdf)

**Comment by pdebuyl** [comment]
@BartoszBartmanski Some references still don't appear correctly, as you can only use one bib reference per square brackets. Can you fix that?

**Comment by BartoszBartmanski** [comment]
Sorry, I did not realise that it was one reference per square brackets. I just updated the repository with those changes. Thank you for clarifying that.

@whedon generate pdf

**Comment by danielskatz** [comment]
Hi
1 - you can have multiple references inside one set of square brackets, but they need to separated with ;s, not ,s - they may also not support a space after the ; but I am not sure.  Try [@a;@b] for example.
2 - commands to whedon need to be the very start of a comment

**Comment by pdebuyl** [comment]
Thank you @danielskatz 

Sorry @BartoszBartmanski , it is indeed better to have the grouped citations as originally intended.

**Comment by BartoszBartmanski** [comment]
Thank you @danielskatz for both suggestions

@danielskatz I just updated the citations in the repository. Let's see whether it's fixed

**Comment by BartoszBartmanski** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02293/joss.02293/10.21105.joss.02293.pdf)

**Comment by BartoszBartmanski** [comment]
@pdebuyl That seems to have fixed that issue

**Comment by CFGrote** [comment]
still missing the target audience.

**Comment by CFGrote** [comment]
Indentation in the README is off at "## Example"

**Comment by BartoszBartmanski** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02293/joss.02293/10.21105.joss.02293.pdf)

**Comment by BartoszBartmanski** [comment]
@CFGrote Thank you for spotting Examples section misalignment. A typo in the README caused this section title to be interpreted as a code block. I also added target audience to both the README and the docs in the most recent commit.

**Comment by CFGrote** [comment]
@BartoszBartmanski, very good. I'm happy with the repo now and recommend publication as is.

**Comment by CFGrote** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by CFGrote** [comment]
@pdebuyl I'm done with the review, my recommendation is: publish as is. Thanks for the experience, this was my first JOSS review.

**Comment by pdebuyl** [comment]
@CFGrote thank you for the review and the feedback!

I will take care of the paper for publication after going through the editor's checklist.

**Comment by pdebuyl** [comment]
@whedon generate pdf

**Comment by pdebuyl** [comment]
@whedon check references

**Comment by whedon** [comment]
PDF failed to compile for issue #2293 with the following error: 

 sh: 0: getcwd() failed: No such file or directory
pandoc: 10.21105.joss.02293.pdf: openBinaryFile: does not exist (No such file or directory)
Looks like we failed to compile the PDF

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1371/journal.pcbi.1005387 is OK

MISSING DOIs

- https://doi.org/10.1186/1752-0509-6-76 may be missing for title: URDME: a modular framework for stochastic simulation of reaction-transport processes in complex geometries
- https://doi.org/10.1093/bioinformatics/bts584 may be missing for title: MesoRD 1.0: Stochastic reaction-diffusion simulations in the microscopic limit
- https://doi.org/10.1371/journal.pcbi.1004923 may be missing for title: Stochastic simulation of biomolecular networks in dynamic environments
- https://doi.org/10.1063/1.4975167 may be missing for title: Reaction rates for reaction-diffusion kinetics on unstructured meshes
- https://doi.org/10.1007/s40571-015-0082-3 may be missing for title: Simulating tissue mechanics with agent-based models: concepts, perspectives and some novel results
- https://doi.org/10.1137/070705039 may be missing for title: The reaction-diffusion master equation as an asymptotic approximation of diffusion to a small target
- https://doi.org/10.1063/1.4816377 may be missing for title: A convergent reaction-diffusion master equation

INVALID DOIs

- None
```

**Comment by pdebuyl** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02293/joss.02293/10.21105.joss.02293.pdf)

**Comment by pdebuyl** [comment]
@BartoszBartmanski can you add the doi for all references in the bib file?

**Comment by BartoszBartmanski** [comment]
@whedon generate pdf

**Comment by BartoszBartmanski** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1186/1752-0509-6-76 is OK
- 10.1093/bioinformatics/bts584 is OK
- 10.1371/journal.pcbi.1004923 is OK
- 10.1063/1.4975167 is OK
- 10.1371/journal.pcbi.1005387 is OK
- 10.1007/s40571-015-0082-3 is OK
- 10.1137/070705039 is OK
- 10.1063/1.4816377 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2293 with the following error: 

 sh: 0: getcwd() failed: No such file or directory
sh: 0: getcwd() failed: No such file or directory
pandoc: 10.21105.joss.02293.pdf: openBinaryFile: does not exist (No such file or directory)
Looks like we failed to compile the PDF

**Comment by BartoszBartmanski** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02293/joss.02293/10.21105.joss.02293.pdf)

**Comment by BartoszBartmanski** [comment]
@pdebuyl I've just updated the reference file with doi

**Comment by pdebuyl** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02293/joss.02293/10.21105.joss.02293.pdf)

**Comment by pdebuyl** [comment]
@BartoszBartmanski thank you for the merges.

Could you make a tagged release and archive it (on Zenodo). When you've done that, report the version number and archive DOI here please. Please make sure the archive deposit has the correct metadata (title and author list) to match the paper.

If you could also confirm the version number, it's currently v2.0.27 but was updated to 2.0.29

**Comment by BartoszBartmanski** [comment]
@pdebuyl The version number is 2.0.29 and DOI is 10.5281/zenodo.3899466

**Comment by pdebuyl** [comment]
@whedon set 2.0.29 as version

**Comment by whedon** [comment]
OK. 2.0.29 is the version.

**Comment by pdebuyl** [comment]
@whedon set 10.5281/zenodo.3899466 as version

**Comment by whedon** [comment]
OK. 10.5281/zenodo.3899466 is the version.

**Comment by pdebuyl** [comment]
@whedon accept

**Comment by whedon** [comment]
No archive DOI set. Exiting...

**Comment by pdebuyl** [comment]
@whedon set 10.5281/zenodo.3899466 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3899466" target="_blank">10.5281/zenodo.3899466</a> is the archive.

**Comment by pdebuyl** [comment]
@whedon set 2.0.29 as version

**Comment by whedon** [comment]
OK. 2.0.29 is the version.

**Comment by pdebuyl** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1371/journal.pcbi.1005220 is OK
- 10.1186/1752-0509-6-76 is OK
- 10.1093/bioinformatics/bts584 is OK
- 10.1371/journal.pcbi.1004923 is OK
- 10.1063/1.4975167 is OK
- 10.1371/journal.pcbi.1005387 is OK
- 10.1007/s40571-015-0082-3 is OK
- 10.1137/070705039 is OK
- 10.1063/1.4816377 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1495

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1495, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by BartoszBartmanski** [comment]
Thank you @pdebuyl @CFGrote @mbkumar for your time and your comments.

This was my first submission to JOSS and it was a very pleasant experience submitting a paper.

**Comment by BartoszBartmanski** [comment]
@pdebuyl Can I ask who has to run the command "whedon accept deposit=true" for the pull request to be merged with joss (https://github.com/openjournals/joss-papers/pull/1495)?

**Comment by pdebuyl** [comment]
@BartoszBartmanski it is also my experience that JOSS reviews are nicer than typical ones :-) Thank you for the feedback.

You don't have to perform any task, nor do I. The "deposit=true"  action is for someone from @openjournals/joss-eics

**Comment by BartoszBartmanski** [comment]
@pdebuyl Is it possible to change the zenodo doi for this submission? 

There was a mix up with some people at zenodo and the archive has been deleted. I can set up a new archive and give the new doi. Would that be ok?

Sorry for the trouble.

**Comment by danielskatz** [comment]
Yes, that's fine - please go ahead

**Comment by danielskatz** [comment]
once you do, I will do the final processing, and I'll make a final check of the paper now

**Comment by danielskatz** [comment]
👋 @pdebuyl  - this paper is too long, which I'm going to let go at this point, but for future papers you edit, a lot of the examples should be in the documentation, with the paper just pointing to the documentation, not including this much detail.  See https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain

**Comment by BartoszBartmanski** [comment]
@danielskatz  the new doi is 10.5281/zenodo.3901670
Sorry for the trouble

**Comment by danielskatz** [comment]
@whedon set 10.5281/zenodo.3901670 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3901670" target="_blank">10.5281/zenodo.3901670</a> is the archive.

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1274070251089670147) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1504
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02293](https://doi.org/10.21105/joss.02293)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by danielskatz** [comment]
Thanks to @CFGrote & @mbkumar for reviewing!

And @pdebuyl for editing!

And congratulations to @BartoszBartmanski (Bartosz Bartmanski) and co-author!!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02293/status.svg)](https://doi.org/10.21105/joss.02293)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02293">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02293/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02293/status.svg
   :target: https://doi.org/10.21105/joss.02293
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02293/status.svg)](https://doi.org/10.21105/joss.02293)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by BartoszBartmanski** [comment]
@danielskatz @pdebuyl @mbkumar @CFGrote Thank you all for your time!

**Comment by pdebuyl** [comment]
@danielskatz thank you for the finalization. I will keep the length in mind for my future editorial assignments.
